server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java

```diff
@@ -267,7 +267,34 @@ private T create0(Author author, String name, long creationTimeMillis) {
 
     @Override
     public Set<String> listRemoved() {
-        return listWithSuffix(SUFFIX_REMOVED);
+        ensureOpen();
+        final File[] files = rootDir.listFiles();
+        if (files == null) {
+            return ImmutableSet.of();
+        }
+
+        Arrays.sort(files);
+        final Builder<String> builder = ImmutableSet.builder();
```

Comment:
nit: How about using `ImmutableSet.Builder`. Importing `Builder` may cause pain when we start to use other types like `ImmutableList` later.

Reply:
That is a nice idea. The implementation has been changed a little so I chose `ImmutableMap.Builder`.